receiver/awslambdareceiver/README.md

```diff
@@ -36,34 +32,61 @@ The `awslambdareceiver` operates as follows:
 1. Accepts Lambda Invocations
 2. Identifies the event source (S3, CloudWatch, etc.)
 3. Uses configured encoding extensions to parse the data
-4. Creates OpenTelemetry log records
-5. Passes parsed logs to the collector pipeline
+4. Creates OpenTelemetry records matching the data type (logs, metrics)
+5. Forward derived records to the next component in the pipeline (processors, exporters)
+
+## Event handling
+
+The receiver automatically detects the event source based on the Lambda invocation message format.
+Sections below describe various details about event handling by this component.
+
+### Supported event types
+
+- **Logs**: Supported through S3 and CloudWatch Logs Subscription Filters
+- **Metrics**: Supported through S3
+
+> [!IMPORTANT]  
+> Currently, metrics mode will always use `awscloudwatchmetricstreams_encoding` extension regardless of the `encoding` parameter set in the receiver configuration.
+> Please make sure ingesting metrics can be decoded using this extension.
```

Comment:
Should we error out if `encoding` is set then, when the receiver is used in a metrics pipeline? Maybe in the future we can allow it to be set to decode metrics stored in some other format, e.g. OTLP protobuf

Reply:
I agree